lona/app.py

```diff
@@ -72,40 +83,71 @@ def resolve_path(self, path):
         )
 
     # decorator ###############################################################
-    def route(self, *args, **kwargs):
-        def decorator(view_class):
+    def route(
+            self,
+            # 1 = lona.MATCH_ALL  https://github.com/python/mypy/issues/10026
+            raw_pattern: Union[str, Literal[1]],
+            name: str = '',
+            interactive: bool = True,
+            http_pass_through: bool = False,
+            frontend_view: Union[None, str, LonaView] = None,
+    ) -> Callable[[Type[LonaView]], None]:
+        def decorator(view_class: Type[LonaView]) -> None:
             self.routes.append(
                 Route(
-                    *args,
+                    raw_pattern=raw_pattern,
                     view=view_class,
-                    **kwargs,
+                    name=name,
+                    interactive=interactive,
+                    http_pass_through=http_pass_through,
+                    frontend_view=frontend_view,
                 )
             )
 
         return decorator
 
-    def middleware(self, middleware_class=None):
-        def decorator(middleware_class):
+    @overload
+    def middleware(self) -> Callable[[Type], None]: ...
+
+    @overload
+    def middleware(self, middleware_class: Type) -> None: ...
+
+    def middleware(self, middleware_class: Optional[Type] = None) -> Optional[Callable[[Type], None]]:  # NOQA: E501
+        def decorator(middleware_class: Type) -> None:
             self.settings.MIDDLEWARES.append(middleware_class)
 
         if callable(middleware_class):
-            return decorator(middleware_class)
+            decorator(middleware_class)
+            return None
 
         else:
             return decorator
 
-    def frontend_view(self, view_class=None):
-        def decorator(view_class):
+    @overload
+    def frontend_view(self) -> Callable[[Type[LonaView]], None]: ...
```

Comment:
@maratori I don't like the one-line syntax. Could you change it to two separate lines?

Reply:
Sure, fixed.